openslides/users/static/js/users/site.js

```diff
@@ -850,76 +826,117 @@ angular.module('OpenSlidesApp.users.site', ['OpenSlidesApp.users'])
 
 .controller('GroupListCtrl', [
     '$scope',
+    '$http',
+    'operator',
     'Group',
-    function($scope, Group) {
+    'permissions',
+    'gettext',
+    'Agenda',
+    'Assignment',
+    'Mediafile',
+    'Motion',
+    'User',
+    function($scope, $http, operator, Group, permissions, gettext, Agenda, Assignment, Mediafile, Motion, User) {
         Group.bindAll({}, $scope, 'groups');
+        $scope.permissions = permissions;
+
+        $scope.apps = [];
+        var lastApp = '';
+
+        // Create the main clustering with appname->permissions
+        angular.forEach(permissions, function(perm) {
+            var permissionApp = perm.value.split('.')[0]; // get appname
+            var translations = { 'agenda': Agenda.verboseName,
+                                 'assignments': Assignment.verboseName,
+                                 'core': gettext('General'),
+                                 'mediafiles': Mediafile.verboseName,
+                                 'motions': Motion.verboseName,
+                                 'users': User.verboseName };
+
+            var translation = translations[permissionApp];
+            if (translation) {
+                permissionApp = translation; // use translation if it is defined
+            } else {  // else: capitalize first char and use app name
+                permissionApp = permissionApp.charAt(0).toUpperCase() + permissionApp.slice(1);
+            }
+
+            if (permissionApp != lastApp) { // new cluster for the app of permission
+                var appCluster = { app_name: permissionApp,
+                                   app_visible: true,
+                                   permissions: [] };
+                lastApp = permissionApp;
+                $scope.apps.push(appCluster);
+            }
+
+            $scope.apps[$scope.apps.length - 1].permissions.push(perm);
+        });
+
+        // sort each app: first all permission with 'see', then 'manage', then the rest
+        // save the permissions in different lists an concat them in the right order together
+        angular.forEach($scope.apps, function (app) {
+            var see = [];
+            var manage = [];
+            var others = [];
+            angular.forEach(app.permissions, function (perm) {
+                if (perm.value.indexOf('see') > -1) {
+                    see.push(perm);
+                } else if (perm.value.indexOf('manage') > -1) {
+                    manage.push(perm);
+                } else {
+                    others.push(perm);
+                }
+            });
```

Comment:
It's a good suggestion and I was thinking about that, but to generate this order, it must be hard-coded. First I didn't want to do that, because when adding a permission, it must (maybe) added in the coded order. I will try to do it as flexible as possible.